docker/smqtk_services.run_images.sh

```diff
@@ -38,57 +30,53 @@ done
 # Argument error checking
 if [ -z "${IMAGE_DIR}" ]
 then
-    echo "ERROR: No image directory provided. Please provide -i|--images option."
+    (
+        echo -n "ERROR: No image directory provided."
+        echo -n " Please provide -i|--images option."
+        echo
+    ) >&2
+
     exit 1
 elif [ -z "$(ls "${IMAGE_DIR}")" ]
 then
-    echo "ERROR: Nothing in provided image directory: ${IMAGE_DIR}"
+    echo "ERROR: Nothing in provided image directory: ${IMAGE_DIR}" >&2
     exit 2
 fi
 
-if [ -z "${DOCKER_NETWORK}" ]
-then
-    echo "ERROR: No docker network name given"
-    exit 3
-fi
-
-#
-# Make sure docker network exists / is created
-#
-if [ -z "$(docker network ls 2>/dev/null | tail -n +2 | grep "${DOCKER_NETWORK}")" ]
-then
-    # network doesn't exist yet, create it
-    echo "Creating new docker network \"${DOCKER_NETWORK}\" using bridge driver"
-    docker network create --driver bridge "${DOCKER_NETWORK}"
-else
-    echo "Found docker network \"${DOCKER_NETWORK}\" already exists"
-fi
-
-#
-# Start and Initialize PostgreSQL container
-#
-echo "Starting up PostgreSQL docker"
-docker run -d --name "${DOCKER_POSTGRES}" --net=${DOCKER_NETWORK} postgres
+export IMAGE_DIR
+mkdir -p smqtk_logs
+docker-compose up -d postgres
 
 # Wait until PSQL instance is up and running by poking psql
-echo "Waiting for a responsive database"
-q=""
 trigger="DatabaseNowResponsive"
-while [ -z "$q" ]
-do
+poll_command='docker-compose exec postgres psql postgres postgres'
+poll_command="$poll_command -c \"\\\\echo $trigger\" 2>/dev/null"
+poll_command="$poll_command | grep -q \"$trigger\""
+
+echo "Waiting for a responsive database"
+(
     set +e
-    q="$(docker exec ${DOCKER_POSTGRES} psql postgres postgres -c "\echo ${trigger}" 2>/dev/null | grep "${trigger}")"
-    set -e
-done
-unset q trigger
+    eval "$poll_command"
+    while (("$?")) ; do eval "$poll_command" ; done
+)
 
 # Create new tables in DB, pulling init scripts from SMQTK container
+#
+# workaround https://github.com/docker/compose/issues/3352
+# When above issue is resolved, replace
+#
+#     docker exec -i $(docker-compose ps -q postgres) ...
+#
+# with
+#
+#     docker-compose exec postgres ...
+#
 echo "Creating required tables"
```

Comment:
As part of this effort, it might make sense to clean up our usage of the postgres image. For example, the postgres image has an officially sanctioned way of running scripts on startup, see "How to extend this image" on their [dockerhub page](https://hub.docker.com/_/postgres/).